### PR TITLE
TMA-690 && TMA-633 tests now verify that synchronize users action fai…

### PR DESCRIPTION
**Pull request by Hahihula**
…ls when supplied with unsupported sync_mode param

### CODE REVIEW

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 1]
```diff
@@ -14,9 +14,10 @@ def check_params(specification, params)
           specification.keys.each do |param_name|
             value = params.send(param_name)
             type = specification[param_name][:type]
-
-            if value.nil?
-              if specification[param_name][:opts][:required]
+            if value.blank?
+              if specification[param_name][:opts][:default]
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name]' 6 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name][:opts]' 5 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Calls 'specification[param_name][:opts][:default]' 2 times (DuplicateMethodCall)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Refers to 'specification' more than self (maybe move it to another class?) (FeatureEnvy)

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/711/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/544/ : SUCCESS in 37s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/821/ : SUCCESS in 32s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/266/ : SUCCESS in 22s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/545/ : SUCCESS in 1m 47s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/822/ : SUCCESS in 1m 59s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1188/ : SUCCESS in 1h 13m 26s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/946/ : UNSTABLE in 45m 46s

File: lib/gooddata/lcm/helpers/check_helper.rb [hunk 2]
```diff
@@ -14,9 +14,10 @@ def check_params(specification, params)
           specification.keys.each do |param_name|
             value = params.send(param_name)
             type = specification[param_name][:type]
-
-            if value.nil?
-              if specification[param_name][:opts][:required]
+            if value.nil? || (value.is_a?( String ) && value.empty?)
```
Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Performs a nil-check (NilCheck)

Comment (yenkins) [inline lib/gooddata/lcm/helpers/check_helper.rb]: Style/SpaceInsideParens: Space inside parentheses detected.

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/712/ : SUCCESS in 13s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/546/ : FAILURE in 35s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/823/ : SUCCESS in 33s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-opensource-repository-check/713/ : SUCCESS in 12s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/547/ : SUCCESS in 31s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/824/ : SUCCESS in 30s

**Comment by panjan** [comment]
ok to merge

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://ci.intgdc.com/job/gooddata-ruby-merge-compliance-check/267/ : SUCCESS in 16s
- https://ci.intgdc.com/job/gooddata-ruby-pronto-code-review-zuul/550/ : SUCCESS in 2m 24s
- https://ci.intgdc.com/job/gooddata-ruby-unit-tests-zuul/827/ : SUCCESS in 2m 12s
- https://ci.intgdc.com/job/gooddata-ruby-integration-tests-zuul/1189/ : SUCCESS in 1h 18m 08s
- https://ci.intgdc.com/job/gooddata-ruby-lcm-integration-tests-zuul/947/ : SUCCESS in 53m 20s

**Pull request merged**